src/kixi/hecuba/minorca.clj

```diff
@@ -112,29 +109,34 @@
   [& args]
   (let [{:keys [project-id embed-url username password] :as opts}
         (:options (parse-opts args cli-options))
-        {:keys [mapping-file processed-file]} config-info
-        s3-files (list-files-in-bucket)]
-    (log/info "S3 files:" s3-files)
-    (doseq [f s3-files]
-      (println "> FILE " f)
-      (let [file-info (s3/get-object cred bucket f)
-            input-data (with-open [r (->> (s3/get-object cred bucket f)
-                                          :object-content
-                                          io/reader)]
-                         (pro/file->seq-of-maps r))
-            processed-content (with-open [in-file (io/reader processed-file)]
-                                (vec (csv/read-csv in-file)))
-            data-to-save (conj (vec processed-content)
-                               [(f/unparse fmt (t/now)) (:key file-info) (:bucket-name file-info)
-                                (:object-metadata file-info) (:object-content file-info)])]
-        (println ">> Pre-processing... " f)
-        (pro/write-to-file processed-file data-to-save)
-        (pre-processing input-data
-                        mapping-file project-id
-                        embed-url username password)
-        (println ">> Processing... " f)
-        (upload-measurement-data input-data mapping-file
-                                 embed-url username password)))))
+        {:keys [s3 mapping-file processed-file]} config-info
+        processed-files (map :file_name (pro/file->seq-of-maps processed-file))
+        s3-files (list-files-in-bucket (:cred s3) (:bucket s3))
+        files-to-process (set/difference (set s3-files) (set processed-files))]
+    (log/info "New S3 files:" files-to-process)
+    (if (> (count files-to-process) 0)
+      (doseq [f files-to-process]
+        (println "> FILE " f)
+        (let [file-info (s3/get-object (:cred s3) (:bucket s3) f)
```

Comment:
`{:keys [key bucket-name object-metadata object-content]} (s3/get-object (:cred s3) (:bucket s3) f)`

and then use as appropriate below as I don't think you use file-info directly, just the parts.